openmc/data/photon.py

```diff
@@ -858,6 +913,81 @@ def _add_bremsstrahlung(self):
         self.bremsstrahlung['photon_energy'] = _BREMSSTRAHLUNG['photon_energy']
         self.bremsstrahlung.update(_BREMSSTRAHLUNG[self.atomic_number])
 
+    def _compute_heating(self):
+        r"""Compute heating cross sections (KERMA)
+
+           Photon energy is deposited as energy loss in three reactions:
+           incoherent scattering, pair production and photoelectric effect.
+           The point-wise heating cross section is calculated as:
+
+           .. math::
+               \sigma_{Hx}(E) &= (E - \overline{E}_x(E)) \cdot \sigma_x(E), x \in \left\{I, PP, PE \right\}
+
+               \overline{E}_I(E) &= \frac {\int E' \sigma_I (E,E',\mu) d\mu} {\int \sigma_I (E,E',\mu) d\mu}
+
+               \overline{E}_{PP} &= 2 m_e c^2 = 1.022 \times 10^6 eV
+
+               \overline{E}_{PE} &= E(\text{fluorescent photons})
+
+           The differential cross section representation for incoherent
+           scattering can be found in the theory manual.
+
+        """
+
+        # Determine a union energy grid
+        energy = np.array([])
+        for mt in (504, 515, 517, 522):
+            if mt not in self:
+                warn('Cross sections for MT={} do not exist. The total heating '
+                     'to be calculated is probably incorrect'.format(mt))
+            else:
+                energy = np.union1d(energy, self[mt].xs.x)
+
+        heating_xs = np.zeros_like(energy)
+
+        # Incoherent scattering
+        if 504 in self:
+            rx = self[504]
+            def _dsigma_dmu(mu, E):
+                k = E / MASS_ELECTRON_EV
+                kout = k / (1.0 + k * (1.0 - mu))
+                x = E * np.sqrt(0.5 * (1.0 - mu)) / PLANCK_C
```

Comment:
Use `math.sqrt` instead. If I understand correctly, `np.sqrt` is actually slower on a scalar because of the extra work of checking whether it is an array.